### PR TITLE
Allow users to perform simple arithmetic operations on histogram aggregations

### LINKED ISSUE #11029: Aggregation to allow users to perform simple arithmetic operations on histogram aggregations
This would be a parent reducer which takes as input multiple metrics inside a histogram. The reduce operation would, for each bucket, present the value of each of these metrics to a script whose result would be the aggregation's output for that bucket.

An example use cases for this would be to subtract two series and output the result.

### PR DESCRIPTION
**Pull request by colings86**
A script can be used to calculate a metric to add to a bucket based on other metrics present in the bucket

Closes #11029

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/search/aggregations/pipeline/seriesarithmetic/SeriesArithmeticParser.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                }
+            } else {
+                throw new SearchParseException(context, "Unexpected token " + token + " in [" + reducerName + "].",
+                        parser.getTokenLocation());
+            }
+        }
+
+        if (bucketsPathsMap == null) {
+            throw new SearchParseException(context, "Missing required field [" + BUCKETS_PATH.getPreferredName()
+                    + "] for derivative aggregation [" + reducerName + "]", parser.getTokenLocation());
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/search/aggregations/pipeline/seriesarithmetic/SeriesArithmeticParser.java]: derivative?

File: core/src/main/java/org/elasticsearch/search/aggregations/pipeline/seriesarithmetic/SeriesArithmeticParser.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        }
+
+        if (script == null) {
+            throw new SearchParseException(context, "Missing required field [" + ScriptField.SCRIPT.getPreferredName()
+                    + "] for derivative aggregation [" + reducerName + "]", parser.getTokenLocation());
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/search/aggregations/pipeline/seriesarithmetic/SeriesArithmeticParser.java]: same

**Comment by jpountz** [comment]
LGTM

**Comment by tarunsapra** [comment]
would be nice if a link to the final docs is added

**Pull request merged**